beta-src/src/components/ui/WDUnitController.tsx

```diff
@@ -54,23 +52,7 @@ const WDUnitController: React.FC<UnitControllerProps> = function ({
 
   processNextCommand(commands, commandActions);
 
-  let unitCanInitiateOrder = false;
-  if ("currentOrders" in data) {
-    const { currentOrders } = data;
-    if (currentOrders) {
-      for (let i = 0; i < currentOrders.length; i += 1) {
-        if (currentOrders[i].unitID === meta.unit.id) {
-          unitCanInitiateOrder = true;
-          break;
-        }
-      }
-    }
-  }
-
   const clickAction = (e) => {
```

Comment:
both clickAction and doubleClickAction are nearly identical now only thing that change is the method. Can we consolidate and pass another parameter to this method.